src/server/common/helpers/session-cache/utils.test.js

```diff
@@ -341,65 +357,252 @@ describe('#utils', () => {
       }
     })
 
-    test('should update multiple siteDetails properties in a single operation', () => {
+    test('should update multiple siteDetails properties in a single operation for a single site', () => {
       const existingCache = {
         projectName: 'Test Project',
         siteDetails: [
           {
             coordinatesType: 'file',
-            existingProperty: 'keepThis'
+            fileUploadType: 'kml'
           }
         ]
       }
 
       mockRequest.yar.get.mockReturnValue(existingCache)
 
-      const updates = {
-        uploadedFile: { filename: 'test.kml', status: 'ready' },
-        extractedCoordinates: [{ lat: 50, lng: -1 }],
-        geoJSON: { type: 'FeatureCollection', features: [] },
-        featureCount: 1
+      const mockCoordinateData = {
+        extractedCoordinates: [],
+        geoJSON: {
+          type: 'FeatureCollection',
+          features: [
+            {
+              type: 'Feature',
+              geometry: {
+                type: 'Polygon',
+                coordinates: [
+                  [
+                    [-1.2345, 50.9876],
+                    [-1.2335, 50.9876],
+                    [-1.2335, 50.9886],
+                    [-1.2345, 50.9886],
+                    [-1.2345, 50.9876]
+                  ]
+                ]
+              }
+            }
+          ]
+        }
       }
 
-      const result = updateExemptionSiteDetailsBatch(mockRequest, updates)
+      const result = updateExemptionSiteDetailsBatch(
+        mockRequest,
+        mockStatus,
+        mockCoordinateData,
+        mockS3Location,
+        false
+      )
+
+      const expected = {
+        coordinatesType: 'file',
+        fileUploadType: 'kml',
+        uploadedFile: {
+          filename: 'test-file',
+          status: 'ready',
+          s3Location: {
+            checksumSha256: 'test-checksum',
+            s3Bucket: 'test-bucket',
+            s3Key: 'test-key'
```

Comment:
Could every uploaded file potentially have a different s3 bucket & key? Just wondering why this is stored with each individual file